NumLib/DOF/LocalToGlobalIndexMap.h

```diff
@@ -73,6 +89,14 @@ class LocalToGlobalIndexMap final
 
     std::size_t size() const;
 
+    std::size_t getNumberOfVariables() const { return (_variable_component_offsets.size() - 1); }
+
+    std::size_t getNumberOfVariableComponents(int variable_id) const
+    {
+        assert(static_cast<unsigned>(variable_id) < getNumberOfVariables());
```

Comment:
Minor: I think std::vector already asserts this in debug mode in the line below.